File: 338. Counting Bits/main.cpp
#include <bitset>
#include <vector>

using std::bitset;
using std::vector;

class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> counter(n + 1, 0);

        for (int i = 0; i < 32; i++) {
            for (int j = 0; j <= n; j++) {
                counter[j] += (j >> i) & 1;
            }
        }

        return counter;
    }
};

class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> counter(n + 1, 0);

        for (int i = 0; i <= n; i++) {
            counter[i] = bitset<32>(i).count();
        }

        return counter;
    }
};

class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> counter(n + 1, 0);

        for (int i = 0; i <= n; i++) {
            counter[i] = counter[i >> 1] + (i & 1);
        }

        return counter;
    }
};

File: 338. Counting Bits/main.py
from typing import List


class Solution:
    def countBits(self, n: int) -> List[int]:
        counter = [0] * (n + 1)

        for num in range(0, n + 1):
            n = num
            while n != 0:
                counter[num] += n & 1
                n >>= 1

        return counter


class Solution:
    def countBits(self, n: int) -> List[int]:
        counter = [0] * (n + 1)

        for num in range(0, n + 1):
            counter[num] = num.bit_count()

        return counter


class Solution:
    def countBits(self, n: int) -> List[int]:
        counter = [0] * (n + 1)

        for num in range(0, n + 1):
            counter[num] = bin(num).count("1")

        return counter


class Solution:
    def countBits(self, n: int) -> List[int]:
        counter = [0] * (n + 1)

        for num in range(0, n + 1):
            counter[num] = counter[num >> 1] + (num & 1)

        return counter
